### DISCUSSION TITLE
What about a "Rstest" tool ?

### DISCUSSION THREAD

**Discussion by abenhamdine** [web-infra-dev/rsbuild#3005] [Ideas]
Hi the ByteDance infra team,

We are currently migrating our whole build chain from vite to rsbuild and until now the experience was a delight, so a big thanks ! 
Performance of rsbuild is astonishing, and the configuration was very easy.
We also have dropped nextra in favor of rspress at the same time, and use storybook with rsbuild

The only remaining tool of our former tooling is vitest : vitest is a great tool, with a pleasant ui.
But because of that, we have to keep a vite configuration, vite plugins etc... to be able to run vitest.  
In a perfect world, we would like to have a tool ("rstest" ? 😉 ) which run our tests relying on our rsbuild configuration.

What do you think of this idea ?
Is it part of your plans ? 
It would be great !

**Comment by chenjiahan** [comment]
Thank you for your support of our tools, we are very honored 😄

We love Vitest and have no plans to start the Rstest project anytime soon. This allows us to focus on existing tools and provide a better experience.

I agree that using Rsbuild and Vitest at the same time brings the maintenance costs of having two build configurations, and we can probably reduce this cost by providing tools to convert some of the Rsbuild configurations to Vitest configurations.

In the long run, we see a need for Rstest, but we don't want to build it from scratch. We hope to have some cooperation with Vitest, such as trying to implement Rstest on top of Vitest, instead of implementing it from scratch.

In addition, if there are users in the community who would like to implement Rstest, we look forward to it and will do our best to support it.

**Reply by abenhamdine** [reply]
I see there https://rsdoctor.dev/blog/release/release-note-1_0?#who-is-using that Rstest is mentionned as a WIP.
Is that true ?? it would be awesome !

**Reply by abenhamdine** [reply]
I've also found this mention here : https://rspack.dev/misc/planning/roadmap#introduce-rstest

**Reply by chenjiahan** [reply]
Yes, stay tuned 😄

**Reply by chenjiahan** [reply]
Rstest repo: https://github.com/web-infra-dev/rstest

**Comment by yyx990803** [comment]
@abenhamdine curious to hear why you decided to migrate away from Vite. What is the main issue that led to this decision?

**Reply by abenhamdine** [reply]
The  single but critical issue we ran into was continous OOM failures when building antd 5.16 (and following versions) with vite.
Even with 12 Go Ram.... which is not a option in CI environments. 
We never found a workaround to this issue, thus we decided to try another builder... and then rsbuild came into the light.

**Reply by yyx990803** [reply]
Got it! Good to know this is going to be fixed for sure with Rolldown.

**Reply by abenhamdine** [reply]
more generaly OOM with vite are an issue for many users , see https://github.com/vitejs/vite/issues/2433

**Reply by gfox1984** [reply]
I'm migrating to Rspack & Rsbuild instead of Vite because of 1st class support for Module Federation, and also the update from webpack & CRA is very straightforward. 

I'm currently trying to add Vitest to my project, but I'm struggling with a lot of issues (eg: `Error: require() of ES Module xxx not supported. Instead change the require of xxx to a dynamic import() which is available in all CommonJS modules`.)

**Reply by ddavid93** [reply]
@yyx990803 we also migrated all projects (15) from Vite to Rsbuild because of SystemJS compatibility (we have a micro-frontend architecture).
Since Vitejs uses esbuild in dev, we weren't able to use it. So, with Rolldown will be that possible?

**Reply by ddavid93** [reply]
> @yyx990803 we also migrated all projects (15) from Vite to Rsbuild because of SystemJS compatibility (we have a micro-frontend architecture). Since Vitejs uses esbuild in dev, we weren't able to use it. So, with Rolldown will be that possible?

@yyx990803 ??
